src/main.rs

```diff
@@ -13,18 +17,64 @@ fn main() {
     }
 }
 
-fn run(mut args: impl Iterator<Item = String>) -> Result<()> {
-    while let Some(arg) = args.next() {
-        if arg == "--" {
-            break;
-        }
+
+async fn relay_signals(child: i32) -> Result<()> {
+    // All termination signals that can be caught
+    let mut signals : Signals = Signals::new(&[
+        Signal::Term,
+        Signal::Int,
+        Signal::Quit,
+        Signal::Hup,
+    ])?;
+
+    while let Some(signal) = signals.next().await {
+      let kill = Command::new("kill")
+          .args(["-s", &(signal.unwrap() as i32).to_string(), &child.to_string()])
+          .stderr(Stdio::null())
+          .stdout(Stdio::null())
+          .spawn()?.wait().await;
+      match kill {
+          Ok(_) => { continue; }
+          Err(_) => { break; }
+      }
```

Comment:
Maybe we want to use the result for all errors in this function?

```suggestion
      Command::new("kill")
          .args(["-s", &(signal? as i32).to_string(), &child.to_string()])
          .stderr(Stdio::null())
          .stdout(Stdio::null())
          .spawn()?.wait().await?;
```